proposals/20200504-falco-artifacts-scope.md

```diff
@@ -0,0 +1,191 @@
+
+Inspired from [#1114](https://github.com/falcosecurity/falco/issues/1114) and many of the weekly community calls.
+
+## Terms
+
+**falco** 
+
+*Standalone Falco binary*
+
+**driver**
+
+*System call provider from the Linux kernel. Either (`bpf`, `module`, or `ptrace`)*
+
+**falco-driver-loader-script**
+
+
+*The bash script found [here](https://github.com/falcosecurity/falco/blob/master/scripts/falco-driver-loader) that tries to compile else download the kernel module.*
+
+**package**
+
+*An installable artifact that is operating system specific. All packages MUST be hosted on bintray for X86_64.*
+
+**image**
+
+*OCI compliant container image hosted on dockerhub with tags for every release.*
+ 
+
+# Packages
+
+Official packages for X86_64 only. All package names should contain a version which is not specified below.
+
+The following convention MUST be used for all packages.
+
+_If a package installs Falco as a daemon (systemd, init) it MUST contain `daemon`._
+
+_If a package installs the Falco kernel module it MUST contain `module`._
+
+_If a package installs the Falco BPF probe it MUST contain `bpf`._
+
+_If a package installs the Falco kernel module it MUST contain `module`._
+
+---
+
+### .deb
+
+ Falco running in debian like systems that will default to the kernel module.
+
+- falco-x86.deb
+     - `falco` only (default depends on `falco-module`)
+ - falco-daemon-x86.deb
+     - `falco` and script to configure as daemon (default depends on `falco-module`)
+ - falco-module-x86.deb
+     - `module` only
+ - falco-bpf-x86.deb
+     - `bpf` only
+ - falco-ptrace-x86.deb
+     - `ptrace` only
+
+### .rpm
+
+ Falco running in rpm like systems that will default to the kernel module.
+
+- falco-x86.deb
+     - `falco` only (default depends on `falco-module`)
+ - falco-daemon-x86.deb
+     - `falco` and script to configure as daemon (default depends on `falco-module`)
+ - falco-module-x86.deb
+     - `module` only
+ - falco-bpf-x86.deb
+     - `bpf` only
+ - falco-ptrace-x86.deb
+     - `ptrace` only
+
+
+### .tar.gz
+
+ - falco-src-x86.tar.gz
+     - No binaries
+     - `INSTALL` file
+ - falco-driver-src-x86.tar.gz
+     - `falco` and `/driver` with `Makefile`
+     - `INSTALL` file
+ - falco-driver-download-x86.tar.gz
+     - `falco` and `falco-driver-loader` download only
+ - falco-driver-full-x86.tar.gz
+     - `falco` and `falco-driver-loader` download else compile
+
+
+# Images
+
+The following convention MUST be used for all container images.
+
+_If a container image requires `--privileged` the name MUST have the `priviliged-` prefix._
+
+_If a container image is for development purproses it MUST have the `dev-` prefix._ 
+
+_If a container image is used to install something, then exit it MUST have the `-install` suffix.*_
+
+---
+
+ - falcosecurity/falco:TAG
+     - Runs `falco` userspace only
+ - falcosecurity/privileged-driver-install:TAG
+     - Runs `falco-driver-loader` and exit
```

Comment:
I think the name `driverloader` is a better name. Maybe we could change it to `falcosecurity/driverinstaller`.

But the fact that it has to be run with `--privileged` regards its usage, not its definition.

AFAIK, there are almost no docker images on the Docker hub that follow this naming pattern. So this could be even more confusing to our users, IMHO.